packages/eslint-plugin/docs/rules/no-misused-promises.mdx

```diff
@@ -100,6 +100,8 @@ The following options are supported:
 
 - `arguments`: Disables checking an asynchronous function passed as argument where the parameter type expects a function that returns `void`
 - `attributes`: Disables checking an asynchronous function passed as a JSX attribute expected to be a function that returns `void`
+- `heritageTypes`: Disables checking an asynchronous method in a type whose extended/implemented heritage type expects that method to return `void`
+  - **Limitation:** For now, this option only checks named methods: that is, call/construct/index signatures are ignored. Call signatures are not required in TypeScript to be consistent with one another, and construct signatures cannot be `async` in the first place. Index signature checking may be implemented in the future.
```

Comment:
[Docs] This is a lot of seemingly-required reading for a limitation that most projects won't hit. It'd be better to move it later down into a `heritageTypes`-specific section.

This rule's docs are a little older than our standard and don't have a section for each of the sub-options... but this is a good opportunity to start one! Request: could you make a `checksVoidReturn` section that at mentions the sub-option names and explains `heritageTypes`?

I think explaining the other sub-options would be best as a follow-up issue. That we we don't bog this PR down with docs nits on `arguments`, `attributes`, etc.

Reply:
Yeah that makes sense, it'd be nice to have separate sections for each suboption.

It looks like the explanations for `checksVoidReturn` and `checksSpreads` were accidentally placed under the `## Examples` section instead of `## Options`. So I've just gone ahead and moved them up in the doc under `## Options` (and turned each of these `checksVoidReturn.<suboption>` bullet points into their own subsections). But let me know if you had/have something else in mind here!